nativescript-background-http/background-http.ios.ts

```diff
@@ -162,7 +184,7 @@ class Session implements common.Session {
 }
 
 class Task extends Observable implements common.Task {
-	private static _tasks = new WeakMap<NSURLSessionTask, Task>();
```

Comment:
Can you elaborate on the problem this WeakMap is fixing?

The NSURLSessionTask key can be GC-ed if nothing else keeps reference to it. The Task has _task field that keeps reference to the NSURLSessionTask. The NSURLSessionTask should be alive as long as one is holding the Task.

If the users add event listeners on the task but loose reference of the Task GC will collect the task and the NSURLSessionTask, and events will stop fire. Is this the problem?

If so, won't making this strong map keep NSURLSessionTasks that errored out.

Won't it be easier to make the WeakMap just Map? It won't require GUIDs. You can still remove key-value pairs from a Map, but you won't worry about IDs.